### DISCUSSION TITLE
How to manager multiple states in a class?

### DISCUSSION THREAD

**Discussion by cruvie** [rrousselGit/riverpod#3282] [Q&A]
I know I can still use this to manage multiple states in a class
```dart
final ssMessageProvider = ChangeNotifierProvider<SSLogicMessage>((ref) {
  return SSLogicMessage();
});

class SSLogicMessage extends ChangeNotifier {
int var1 = 20;
List<int> var2 =[];
}
```
but riverpod does not recommend this way, now.

I have many states to use, should I make them all global variables? that's not good Because these variables are likely to conflict because of the same name, giving them different names is cumbersome.

anyone can help me out?

**Comment by XuanTung95** [comment]
I don't know the answer but I wonder did Riverpod explains why ChangeNotifier and StateNotifier are not recommended somewhere? I feel completely fine using those providers.

**Reply by XuanTung95** [reply]
Oops, after re-checking the docs, I found that the new syntax


     @override 
     MyState build() => MyState();

 is just like `StateNotifier<MyState>` but with less code. If I don't want to break MyState into smaller providers, I still can use it the same as before. 
 
I can use the new syntax to reduce some lines of code without breaking too much.
